src/documents/serialisers.py

```diff
@@ -527,6 +602,21 @@ def update(self, instance: Document, validated_data):
             instance.save()
         if "created_date" in validated_data:
             validated_data.pop("created_date")
+        if instance.custom_fields.count() > 0 and "custom_fields" in validated_data:
+            for custom_field_instance in instance.custom_fields.filter(
+                field__data_type=CustomField.FieldDataType.DOCUMENTLINK,
+            ):
+                if custom_field_instance.field not in [
+                    field["field"] for field in validated_data["custom_fields"]
+                ]:
+                    # Doc link field is being removed entirely
+                    for doc_id in custom_field_instance.value:
+                        CustomFieldInstanceSerializer.remove_doclink(
+                            None,
```

Comment:
If you make `remove_doclink` an `@staticmethod` you wouldn't need to pass None here.  As it doesn't use `self`, that is fine to do.

Reply:
Thanks!